src/App.tsx

```diff
@@ -1,61 +1,98 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { FormEventHandler, useState } from 'react';
+import usersFromServer from './api/users';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [userError, setUserError] = useState('');
+
+  const handleSubmit: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    const foundUser = usersFromServer.find(user => user.name === selectedUser);
+    const newId = Math.max(...todosFromServer.map(todo => todo.id));
+
+    if (!title) {
+      setTitleError('Please enter a title');
+    }
+
+    if (!selectedUser) {
+      setUserError('Please choose a user');
+    }
+
+    if (title && selectedUser && foundUser) {
+      todosFromServer.push({
+        id: newId + 1,
+        title,
+        completed: false,
+        userId: foundUser.id,
+      });
+
+      setTitle('');
+      setSelectedUser('');
+    }
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    setTitleError('');
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(event.target.value);
+
+    setUserError('');
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form onSubmit={handleSubmit} action="/api/todos" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
+          <input
+            type="text"
+            id="title"
+            placeholder="Enter a title"
+            data-cy="titleInput"
+            value={title}
+            onChange={handleTitle}
+          />
+          <span className="error">{titleError}</span>
```

Comment:
Render this span conditionally so it doesn't get rendered when there is no error

Reply:
Same for user error